### PR TITLE
Add synchronization for scaling

### PR DESCRIPTION
**Pull request by hxdtest**

### CODE REVIEW

File: dlrover/trainer/tensorflow/util/estimator_util.py [hunk 1]
```diff
@@ -151,6 +151,13 @@ def _unique(hooks):
             key,
             [c.__class__.__name__ for c in hooks],
         )
+
+        logger.info(
+            "Appending hooks after deduplication: %s = %s",
```
Comment (hxdtest) [inline dlrover/trainer/tensorflow/util/estimator_util.py:156]: To be deleted.

File: dlrover/python/master/servicer.py [hunk 2]
```diff
@@ -366,6 +367,7 @@ def update_node_addr(self, reqeuest, _):
 
         response = elastic_training_pb2.Response()
         response.success = True
+        response.reason = ""
```
Comment (hxdtest) [inline dlrover/python/master/servicer.py:370]: To be deleted.

File: dlrover/python/master/servicer.py [hunk 3]
```diff
@@ -432,11 +434,13 @@ def join_sync(self, request, _):
         res.success = self._sync_service.join_sync(
             request.sync_name, request.worker_type, request.worker_id
         )
+        res.reason = ""
```
Comment (hxdtest) [inline dlrover/python/master/servicer.py:437]: To be deleted.

File: dlrover/python/master/servicer.py [hunk 4]
```diff
@@ -432,11 +434,13 @@ def join_sync(self, request, _):
...
         return res
 
     def sync_finished(self, request, _):
         res = elastic_training_pb2.Response()
         res.success = self._sync_service.sync_finished(request.sync_name)
+        res.reason = ""
```
Comment (hxdtest) [inline dlrover/python/master/servicer.py:443]: To be deleted.

File: dlrover/python/master/servicer.py [hunk 5]
```diff
@@ -447,6 +451,7 @@ def barrier(self, request, _):
             )
         else:
             res.success = self._sync_service.barrier(request.barrier_name)
+        res.reason = ""
```
Comment (hxdtest) [inline dlrover/python/master/servicer.py:454]: To be deleted.

File: dlrover/trainer/tensorflow/executor/estimator_executor.py [hunk 6]
```diff
@@ -181,13 +181,23 @@ def _prepare_estimator_config_and_params(self):
             TFConstants.CheckpointIncrementalSaveSecs.name, None
         )
         if is_tf_115() and checkpoint_incremental_save_secs:
+            CheckpointSaverHook.after_run = ck_after_run
+            logger.info(
+                "CheckpointSaverHook.after_run {}".format(
+                    CheckpointSaverHook.after_run.__code__
```
Comment (hxdtest) [inline dlrover/trainer/tensorflow/executor/estimator_executor.py:187]: To be deleted.

File: dlrover/trainer/tensorflow/executor/estimator_executor.py [hunk 7]
```diff
@@ -181,13 +181,23 @@ def _prepare_estimator_config_and_params(self):
...
             checkpoint_save_hook = CheckpointSaverHook(
                 self._model_dir,
                 save_steps=save_steps,
                 save_secs=save_secs,
                 incremental_save_secs=checkpoint_incremental_save_secs,
             )
-            checkpoint_save_hook.after_run = ck_after_run
+            logger.info(
+                "CheckpointSaverHook.after_run {}".format(
+                    checkpoint_save_hook.after_run.__code__
```
Comment (hxdtest) [inline dlrover/trainer/tensorflow/executor/estimator_executor.py:198]: To be deleted.

**Review by workingloong — APPROVED** [review]
LGTM

**Pull request merged**